src/objects/gate.js

```diff
@@ -0,0 +1,36 @@
+import { scaleNearestNeighbor } from "../helperFunctions/imageFunctions.js";
+
+export default class Gate
+{
+	constructor(img)
+	{
+		if (this.constructor === Gate)
+		{
+			throw new TypeError("Abstract class \"Gate\" cannot be instantiated directly.");
+		}
+		this.img = img;
+	}
+
+	answer()
+	{
+		if (this.schema === undefined)
```

Comment:
I'm not able to see where `schema` is being set/should be set - is this a property on all js classes or something?

Reply:
I looked up for a way to make abstract classes in javascript and because there is no official keyword to turn a class abstract, I looked it up and that was what I found.